Feat add in build tool control to config

mackenziedott opened the pull request:
Add in the ability to control build tool for CICD purposes as well as output format (some tools only allow docker specific images) via the config file.

Commit:
Added control of the UC name to the config file.  This allows for more control over the pod-names.

Commit:
handling an issue when dagster-uc is downloaded without build dependencies

Commit:
bumping version

Commit:
Merge branch 'main' of https://github.com/ASML-Labs/dagster-uc into feat-adjust-pod-name-prefix

Commit:
ruff updates

Commit:
updated build_tool to use values from config

Commit:
Merge branch 'main' of https://github.com/ASML-Labs/dagster-uc into feat-add-in-build-tool-control-to-config

Commit:
ruff cleanup

Commit:
Updating Error warning to check for required sudo to run podman

Commit:
ruff format

ion-elgreco commented on dagster_uc/manage_user_code_deployments.py:
```diff
@@ -397,7 +399,13 @@ def deployment_deploy(
+                if not is_command_available(f"sudo {BuildTool.podman.value}"):
+                    raise Exception("Podman installation is required to run dagster-uc.")
+                else:
+                    raise Warning("Sudo is required to run podman")
```
This is not true, you can run podman fine without sudo

mackenziedott replied:
Apparently not in Azure Pipelines! Either that or I have to do some funkier stuff to make it work. Running podman info fails right after installation, but sudo podman info works fine.

ion-elgreco commented:
A while I removed docker support to simplify the maintenance of the project. Additionally the way it's currently reintroduce will causes issues because fetching tags from the repository is only done using podman at  the moment.

Commit:
Removed build tool selection per PR comments. Added more sudo usage.

mackenziedott commented:
> A while I removed docker support to simplify the maintenance of the project. Additionally the way it's currently reintroduce will causes issues because fetching tags from the repository is only done using podman at the moment.

Understood. i have removed that from the PR, I left in the build format as that may be useful later on.

Commit:
added missing sudos

ion-elgreco commented on dagster_uc/utils.py:
```diff
@@ -175,22 +184,37 @@ def build_and_push(
-        ".",
-    ]
+    if build_tool == BuildTool.podman.value and build_format == "docker":
+        cmd = [
```
you can simplify this by making just doing:

```
if build_tool == BuildTool.podman.value and build_format == "docker":
    cmd += ["--format", "docker"]
```

Commit:
Updated per PR,

Commit:
ruff updates

gfvioli commented on dagster_uc/utils.py:
```diff
@@ -134,18 +137,24 @@ def get_azure_access_token(image_registry: str) -> bytes:
     typer.echo(f"Logging into acr with {BuildTool.podman.value}...")
     token = get_azure_access_token(image_registry)
     cmd = [
-        BuildTool.podman.value,
```
I would keep `BuildTool.podman.value`

gfvioli commented on dagster_uc/utils.py:
```diff
@@ -175,22 +184,26 @@ def build_and_push(
     os.chdir(repository_root)
 
     cmd = [
-        BuildTool.podman.value,
```
I would keep BuildTool.podman.value

gfvioli commented on dagster_uc/utils.py:
```diff
@@ -175,22 +184,26 @@ def build_and_push(
     ]
+    if build_tool == BuildTool.podman.value and build_format == "docker":
+        cmd += ["--format", "docker"]
+    cmd += ["."]  # Since this always has to be at the nd
```
end, not nd

Commit:
removed instances of using build_tool

gfvioli commented on dagster_uc/utils.py:
```diff
@@ -200,10 +213,10 @@ def build_and_push(
+        login_registry(image_registry=image_registry, build_tool=build_tool, use_sudo=use_sudo)
 
     typer.echo("Pushing image...")
-    cmd = [BuildTool.podman.value, "push", os.path.join(image_registry, f"{image_name}:{tag}")]
```
I would keep `BuildTool.podman.value`

mackenziedott replied:
Done. removed usage of build_tool as well since no longer needed.

Commit:
cleanout of build tool

Commit:
fixing comment typo
